uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1026.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1026 {
+
+  /**
+   * Deletes Default Groups
+   *
+   * @return bool
+   */
+  public function upgrade_1026() {
+    $this->up1026_deleteDefaultGroups([
+      'Advisory board',
+      'Newsletter Subscribers',
+      'Summer Program Volunteers',
+      'Administrators',
+    ]);
+
+    $this->up1026_disableAndHideDefaultGroup('Case_Resources');
+
+    return TRUE;
+  }
+
+  /**
+   * Deletes the Groups passed by params
+   *
+   * @param array $groupsToDelete
+   */
+  private function up1026_deleteDefaultGroups($groupsToDelete) {
+    $groups = civicrm_api3('Group', 'get', [
```

Comment:
Chained API call will be more appropriate here.  `up1026_deleteDefaultGroups` implementation can be reduced to like 4/5 lines with it. Same with `up1026_disableAndHideDefaultGroup` below. Kindly refactor.